### PR TITLE
Tool 151 rewrite keychain helper

### PR DESCRIPTION
**Pull request by lszucs**
Ruby  implementation was migrated mostly in a mirror translation fashion. Some responsibilities were left for the future caller, e.g. there is no public ImportCertificates method, only a singular one.

Wrote simple tests for main functions, but it there is room for improvement in terms of tests and testability, e.g. using reader and writer interfaces instead of hardcoded stdins and stdouts.

CI was not updated for the Go implementation.

### CODE REVIEW

File: keychain/keychain.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+	return &Keychain{
+		Path:     path,
+		Password: password,
+	}, nil
+}
+
+// ImportCertificate adds the certificate at path, protected by
+// passphrase to the kc keychain.
+func (kc Keychain) ImportCertificate(path, passphrase string) error {
```
Comment (lpusok) [inline keychain/keychain.go]: Would prefer to provide the certificate as a []byte instead of a path.

Comment (lpusok) [reply keychain/keychain.go]: It seems that is possible through the Sec* API calls, but not the security CLI.

Comment (BirmacherAkos) [reply keychain/keychain.go]: I think it's not an option for the `security` to import a []byte array. It waits for the certificate path:

```
Usage: import inputfile [-k keychain] [-t type] [-f format] [-w] [-P passphrase] [options...]
    -k  Target keychain to import into
    -t  Type = pub|priv|session|cert|agg
    -f  Format = openssl|openssh1|openssh2|bsafe|raw|pkcs7|pkcs8|pkcs12|netscape|pemseq
    -w  Specify that private keys are wrapped and must be unwrapped on import
    -x  Specify that private keys are non-extractable after being imported
    -P  Specify wrapping passphrase immediately (default is secure passphrase via GUI)
    -a  Specify name and value of extended attribute (can be used multiple times)
    -A  Allow any application to access the imported key without warning (insecure, not recommended!)
    -T  Specify an application which may access the imported key (multiple -T options are allowed)
Use of the -P option is insecure
```

Comment (BirmacherAkos) [reply keychain/keychain.go]: So I think it's fine to provide the cert's path

**Pull request closed without merging**